games-emulation/retroarch/retroarch-1.7.1.recipe

```diff
@@ -0,0 +1,60 @@
+SUMMARY="The reference frontend of libretro"
+DESCRIPTION="RetroArch is a frontend for emulators, game engines and media players. \
+It enables you to run classic games on a wide range of computers and consoles \
+through its slick graphical interface. It has advanced features like shaders, \
+netplay, rewinding, next-frame response times, and more!"
+HOMEPAGE="https://libretro.com"
+COPYRIGHT="2010-2018 The RetroArch Team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/libretro/RetroArch/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="e9189c7da6b2ca0142528eca14f26104ee322f052f8609203d90b60938d8d3f3"
+SOURCE_DIR="RetroArch-$portVersion"
+PATCHES="retroarch-$portVersion.patchset"
+
+ARCHITECTURES="x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	retroarch$secondaryArchSuffix = $portVersion
+	app:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch_cg2glsl$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libsdl2_2.0$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	ffmpeg$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsdl2$secondaryArchSuffix
+	devel:libgl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:which
+	cmd:git
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	"
+GLOBAL_WRITABLE_FILES="
+	$relativeSettingsDir/retroarch.cfg keep-old
+"
+
+BUILD()
+{
+	./fetch-submodules.sh
+	./configure --prefix=$installDestDir --global-config-dir=$settingsDir --bindir=$binDir --datarootdir=$dataRootDir --docdir=$docDir --mandir=$manDir --with-assets_dir=$dataDir
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+	addAppDeskbarSymlink $binDir/retroarch RetroArch
```

Comment:
```diff
-	addAppDeskbarSymlink $binDir/retroarch RetroArch
+	addAppDeskbarSymlink "$appsDir"/retroarch RetroArch
+	install -T retroarch.cfg $dataDir/retroarch/retroarch.cfg.default
```

Reply:
I hesitated between $binDir and $appDir for that one. Is there a fundamental difference? The docs I've read were a bit vague on this.

Reply:
Also I'm not sure about the .default for the default cfg, as this file really is meant to be read by retroarch as a template when no user-specific cfg file exists.

Reply:
Seems like replacing addAppDeskbarSymlink by $appsDir breaks the Menu link (even though I replaced --bindir properly)

Reply:
Having a template config in `/system/data/retroarch/` is not really required, but it can be handy in case the user makes changes to `/system/settings/retroarch.cfg` and forgets to make a backup of it. (Although the app does not modify `/system/settings/retroarch.cfg`, the user can still change it, and even break it.)
So the template config serves both as a backup config for 1.7.1 and as a reference when 1.7.2 becomes available, so that the auto-merge feature can compute the diff between the template config of 1.7.2 vs 1.7.1 and apply that diff to the user's config in `/system/settings/`.
(Of course if the template config does not change between 1.7.1 and 1.7.2 then the `auto-merge` will behave like a `keep-old`.)

Regarding the `$appsDir` vs `$binDir` the rule is that command line runtimes (with no GUI) should be in `$binDir`, while GUI programs are better in `$appsDir`. But I guess there can be exceptions.
BTW, `/boot/system/apps` is in PATH, so it is even possible to start retroarch from the command line.

Reply:
Sounds great, thanks. I've fixed most of the issues reported, I'm now fiddling with the extra SOURCE_URIs to replace the fetch script.